src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -147,7 +149,7 @@ private static string GetDirectoryName(string path, bool isUnixLike)
             return null;
         }
 
-        internal static bool IsSameDirectoryOrChildOf(string child, string parent)
+        internal static bool IsSameDirectoryOrChildOf([DisallowNull]string? child, string parent)
```

Comment:
```suggestion
        internal static bool IsSameDirectoryOrChildOf([NotNullWhen(true)] string? child, string parent)
```

Reply:
I don't want callers to pass `null`. 
`[NotNullWhen(true)]` is a post-condition only.

---
In reply to: [368191535](https://github.com/dotnet/roslyn/pull/41063#discussion_r368191535) [](ancestors = 368191535)